sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -1335,6 +1446,96 @@ public void onFailure(Call<ResponseBody> call, Throwable t) {
         }
     }
 
+    private CallAndOptionalResult<BlobDeleteResponse> deleteWithRestResponseIntern(String containerName,
+                                                                                   String blobName,
+                                                                                   String snapshot,
+                                                                                   Integer timeout,
+                                                                                   String version,
+                                                                                   String leaseId,
+                                                                                   DeleteSnapshotsOptionType deleteSnapshots,
+                                                                                   OffsetDateTime ifModifiedSince,
+                                                                                   OffsetDateTime ifUnmodifiedSince,
+                                                                                   String ifMatch,
+                                                                                   String ifNoneMatch,
+                                                                                   String requestId,
+                                                                                   Callback<BlobDeleteResponse> callback) {
+        DateTimeRfc1123 ifModifiedSinceConverted = ifModifiedSince == null ? null :
+            new DateTimeRfc1123(ifModifiedSince);
+        DateTimeRfc1123 ifUnmodifiedSinceConverted = ifUnmodifiedSince == null ? null :
+            new DateTimeRfc1123(ifUnmodifiedSince);
+
+        Call<Void> call = service.delete(containerName,
+            blobName,
+            snapshot,
+            timeout,
+            leaseId,
+            deleteSnapshots,
+            ifModifiedSinceConverted,
+            ifUnmodifiedSinceConverted,
+            ifMatch,
+            ifNoneMatch,
+            XMS_VERSION, // TODO: Replace with 'version'.
+            requestId);
+
+        if (callback != null) {
+            executeCall(call, new retrofit2.Callback<Void>() {
+                @Override
+                public void onResponse(@NonNull Call<Void> call, @NonNull Response<Void> response) {
+                    if (response.isSuccessful()) {
+                        if (response.code() >= 200 && response.code() < 300) {
+                            BlobDeleteHeaders typedHeaders = deserializeHeaders(response.headers(),
+                                BlobDeleteHeaders.class);
+
+                            callback.onResponse(new BlobDeleteResponse(response.raw().request(),
+                                response.code(),
+                                response.headers(),
+                                response.body(),
```

Comment:
since we need to change from `Call<Void>` to `Call<ResponseBody>` here instead of `response.body()` it should be `null`.